packages/core/src/internal/utils/identity.ts

```diff
@@ -65,6 +65,31 @@ export function createId(prefix = '_') {
   return `${prefix}${Math.random().toString(36).substr(2, 9)}`;
 }
 
+// simplistic way to test objects for equality
+// note that it ignores/removes methods from objects
+export function objectNaiveDeepEquals(obj1: {}, obj2: {}) {
+  return JSON.stringify(obj1) === JSON.stringify(obj2);
+}
+
+export function getFromObjectPath(path: string, dataObj: any, fallback = `$\{${path}}`) {
```

Comment:
`getFromObjectPath('a.a1', { a: { a1: 'ohai', a2: 'howdy }, b: 'zdrasti' })` will return `"ohai"`. If you tried to find `getFromObjectPath('a.a3', { a: { a1: 'ohai', a2: 'howdy }, b: 'zdrasti' })` it would return `"${a.a3}"` as a string because there is no `a3`.